File: gen_scripts/ruby/scripts/o00.flags.rb
#!/usr/bin/env ruby
usage = <<END

Usage: #{$0} [-c|-e|-l|-k|-p|-m|-t] [-h|-?]

  -c  Coffee
  -e  Espresso
  -l  Latte
  -k  Machiato
  -p  Capucino
  -m  Mocha
  -t  Tea
  -h  Display this help message
  -?  Display this help message

END

case ARGV[0]
when "-c"
  puts "You ordered a Coffee."
  exit 0
when "-e"
  puts "You ordered an Espresso."
  exit 0
when "-l"
  puts "You ordered a Latte."
  exit 0
when "-k"
  puts "You ordered a Machiato."
  exit 0
when "-p"
  puts "You ordered a Capucino."
  exit 0
when "-m"
  puts "You ordered a Mocha."
  exit 0
when "-t"
  puts "You ordered a Tea."
  exit 0
when "-h", "-?"
  print usage
  exit 0
end

$stderr.print usage
exit 1

File: lessons/gen_scripts/python2/scripts/o00.flags.py
#!/usr/bin/env python2
import sys
import getopt

USAGE = """
Usage: %s [-c|-e|-l|-k|-p|-m|-t] [-h|-?]

  -c  Coffee
  -e  Espresso
  -l  Latte
  -k  Machiato
  -p  Capucino
  -m  Mocha
  -t  Tea
  -h  Display this help message
  -?  Display this help message
"""

try:
    opts, args = getopt.getopt(sys.argv[1:], "celkpmth?")
except getopt.GetoptError:
    opts = []

if not opts:
    sys.stderr.write(USAGE % sys.argv[0] + "\n")
    sys.exit(1)

for opt, _ in opts:
    if opt == "-c":
        print "You ordered a Coffee."
        sys.exit(0)
    elif opt == "-e":
        print "You ordered an Espresso."
        sys.exit(0)
    elif opt == "-l":
        print "You ordered a Latte."
        sys.exit(0)
    elif opt == "-k":
        print "You ordered a Machiato."
        sys.exit(0)
    elif opt == "-p":
        print "You ordered a Capucino."
        sys.exit(0)
    elif opt == "-m":
        print "You ordered a Mocha."
        sys.exit(0)
    elif opt == "-t":
        print "You ordered a Tea."
        sys.exit(0)
    elif opt in ("-h", "-?"):
        print USAGE % sys.argv[0]
        sys.exit(0)
